pkg/logs/launchers/integration/launcher.go

```diff
@@ -86,58 +118,138 @@ func (s *Launcher) run() {
 			for _, source := range sources {
 				// TODO: integrations should only be allowed to have one IntegrationType config.
 				if source.Config.Type == config.IntegrationType {
-					logFilePath, err := s.createFile(cfg.IntegrationID)
-					if err != nil {
-						ddLog.Warn("Failed to create integration log file: ", err)
-						continue
+					// This check avoids duplicating files that have already been created
+					// by scanInitialFiles
+					logFile, exists := s.integrationToFile[cfg.IntegrationID]
+
+					if !exists {
+						logFile, err = s.createFile(cfg.IntegrationID)
+						if err != nil {
+							ddLog.Warn("Failed to create integration log file:", err)
+							continue
+						}
+
+						// file to write the incoming logs to
+						s.integrationToFile[cfg.IntegrationID] = logFile
 					}
-					filetypeSource := s.makeFileSource(source, logFilePath)
-					s.sources.AddSource(filetypeSource)
 
-					// file to write the incoming logs to
-					s.integrationToFile[cfg.IntegrationID] = logFilePath
+					filetypeSource := s.makeFileSource(source, logFile.filename)
+					s.sources.AddSource(filetypeSource)
 				}
 			}
 
 		case log := <-s.integrationsLogsChan:
-			logFilePath := s.integrationToFile[log.IntegrationID]
+			s.receiveLogs(log)
+		case <-s.stop:
+			return
+		}
+	}
+}
 
-			err := s.ensureFileSize(logFilePath)
-			if err != nil {
-				ddLog.Warn("Failed to get file size: ", err)
-				continue
-			}
+// receiveLogs handles writing incoming logs to their respective file as well as
+// enforcing size limitations
+func (s *Launcher) receiveLogs(log integrations.IntegrationLog) {
+	fileToUpdate := s.integrationToFile[log.IntegrationID]
 
-			err = s.writeFunction(logFilePath, log.Log)
-			if err != nil {
-				ddLog.Warn("Error writing log to file: ", err)
-			}
-		case <-s.stop:
+	// If there is no file to write to, skip writing to it
+	if fileToUpdate == nil {
+		return
+	}
+
+	// Ensure the individual file doesn't exceed integrations_logs_files_max_size
+	// Add 1 because we write the \n at the end as well
+	logSize := int64(len(log.Log)) + 1
+	if fileToUpdate.size+logSize > s.fileSizeMax {
+		s.combinedUsageSize -= fileToUpdate.size
+		err := s.makeFile(fileToUpdate.filename)
+		if err != nil {
+			ddLog.Warn("Failed to delete and remake oversize file", err)
 			return
 		}
+
+		fileToUpdate.size = 0
+	}
+
+	// Ensure combined logs usage doesn't exceed integrations_logs_total_usage by
+	// deleting files until total usage falls below the set maximum
+	for s.combinedUsageSize+logSize > s.combinedUsageMax {
+		leastRecentlyModifiedFile := s.getLeastRecentlyModifiedFile()
+
+		err := s.deleteFile(leastRecentlyModifiedFile)
+		if err != nil {
+			ddLog.Warn("Error deleting log file:", err)
```

Comment:
I'd also make this an error log. and same for line 186 below.

Reply:
Changed these to error logs 👍